### PR TITLE
Generate phpdoc in php generated files.

### PR DESCRIPTION
**Pull request by TeBoring**

### CODE REVIEW

**Comment by TeBoring** [comment]
Fixes #2135

File: src/google/protobuf/compiler/php/php_generator.cc [hunk 1]
```diff
@@ -759,11 +754,174 @@ void GenerateFile(const google::protobuf::FileDescriptor* file,
...
+  string result = value;
+
+  string::size_type pos = result.find_first_of('\n');
+  if (pos != string::npos) {
+    result.erase(pos);
+  }
+
+  // If line ends in an opening brace, make it "{ ... }" so it looks nice.
+  if (!result.empty() && result[result.size() - 1] == '{') {
+    result.append(" ... }");
```
Comment (michaelbausor) [inline src/google/protobuf/compiler/php/php_generator.cc]: Maybe update test.proto to test this feature, and some of the other escaping features as well.

File: src/google/protobuf/compiler/php/php_generator.cc [hunk 2]
```diff
@@ -759,11 +754,174 @@ void GenerateFile(const google::protobuf::FileDescriptor* file,
...
+
+  return result;
+}
+
+void GenerateMessageDocComment(io::Printer* printer,
+                               const Descriptor* message) {
+  printer->Print("/**\n");
+  GenerateDocCommentBody(printer, message);
+  printer->Print(
+    " * Protobuf type {@code ^fullname^}\n"
```
Comment (TeBoring) [inline src/google/protobuf/compiler/php/php_generator.cc]: Modified to <code>...</code>

File: src/google/protobuf/compiler/php/php_generator.cc (hunk 1 above)
Comment (TeBoring) [inline src/google/protobuf/compiler/php/php_generator.cc]: Updated test.proto.
Php only supports proto3 now. In proto3, there is no case to have '{' at the end of line.

**Comment by michaelbausor** [comment]
Overall this looks good to me. One thing I am unsure about is the best way to handle the code blocks. From the PSR-5 doc, this section (https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#52-description) suggests that markdown is supported. So perhaps we should use ``` instead of \<code> tags? See also this discussion: https://groups.google.com/forum/#!topic/php-fig/5odnVz3NogM

**Comment by TeBoring** [comment]
The mark-down is supported for Description.
However, most of our comments just have one line, which will be regarded as Summary.
In Summary, mark-down is not supported.
https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#51-summary

**Comment by michaelbausor** [comment]
Good point, in that case using code tags seems best.

**Pull request merged**